application/src/main/java/org/opentripplanner/osm/tagmapping/OsmTagMapper.java

```diff
@@ -1,18 +1,731 @@
+    );
+    props.setProperties("highway=service;cycleway=shared_lane", withModes(ALL).bicycleSafety(0.73));
+    props.setProperties(
+      "highway=residential;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(0.77)
+    );
+    props.setProperties(
+      "highway=residential_link;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(0.77)
+    );
+    props.setProperties(
+      "highway=tertiary;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(0.83)
+    );
+    props.setProperties(
+      "highway=tertiary_link;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(0.83)
+    );
+    props.setProperties(
+      "highway=secondary;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(1.25)
+    );
+    props.setProperties(
+      "highway=secondary_link;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(1.25)
+    );
+    props.setProperties("highway=primary;cycleway=shared_lane", withModes(ALL).bicycleSafety(1.75));
+    props.setProperties(
+      "highway=primary_link;cycleway=shared_lane",
+      withModes(ALL).bicycleSafety(1.75)
+    );
+
+    /* cycleway=opposite */
+    props.setProperties(
+      "highway=*;cycleway=opposite",
+      withModes(PEDESTRIAN_AND_BICYCLE).bicycleSafety(1, 1.4)
+    );
+    props.setProperties("highway=service;cycleway=opposite", withModes(ALL).bicycleSafety(1.1));
+    props.setProperties(
+      "highway=residential;cycleway=opposite",
+      withModes(ALL).bicycleSafety(0.98)
+    );
+    props.setProperties(
+      "highway=residential_link;cycleway=opposite",
+      withModes(ALL).bicycleSafety(0.98)
+    );
+    props.setProperties("highway=tertiary;cycleway=opposite", allWayProperties);
+    props.setProperties("highway=tertiary_link;cycleway=opposite", allWayProperties);
+    props.setProperties(
+      "highway=secondary;cycleway=opposite",
+      withModes(ALL).bicycleSafety(1.5, 1.71)
+    );
+    props.setProperties(
+      "highway=secondary_link;cycleway=opposite",
+      withModes(ALL).bicycleSafety(1.5, 1.71)
+    );
+    props.setProperties(
+      "highway=primary;cycleway=opposite",
+      withModes(ALL).bicycleSafety(2.06, 2.99)
+    );
+    props.setProperties(
+      "highway=primary_link;cycleway=opposite",
+      withModes(ALL).bicycleSafety(2.06, 2.99)
+    );
+
+    /*
+     * path designed for bicycles (should be treated exactly as a cycleway is), this is a multi-use path (MUP)
+     */
+    props.setProperties(
+      "highway=path;bicycle=designated",
+      withModes(PEDESTRIAN_AND_BICYCLE).bicycleSafety(0.60)
+    );
+
+    /* special cases for footway, pedestrian and bicycles */
+    props.setProperties(
+      "highway=footway;bicycle=designated",
+      withModes(PEDESTRIAN_AND_BICYCLE).bicycleSafety(0.75)
+    );
+    props.setProperties(
+      new ExactMatchSpecifier(
+        new Condition.Equals("highway", "footway"),
+        new Condition.Equals("bicycle", "yes"),
+        new Condition.Equals("area", " yes")
```

Comment:
```suggestion
        new Condition.Equals("area", "yes")
```

BTW, you can also write it with the string-only notation: https://github.com/opentripplanner/OpenTripPlanner/blob/f4bfedfc41d3b92df45927a72edcab28624510d6/application/src/main/java/org/opentripplanner/osm/tagmapping/HoustonMapper.java#L26

Reply:
Thanks! That is indeed much better way. Updated.